x-pack/platform/plugins/shared/dataset_quality/public/rule_types/degraded_docs/rule_form/rule_form.tsx

```diff
@@ -0,0 +1,327 @@
+
+        if (adHocDataView) {
+          currentDataView = adHocDataView;
+        } else {
+          currentDataView = await dataViews.create({
+            title: ruleParams.name,
+            timeFieldName: '@timestamp',
+          });
+
+          setAdHocDataViews((prev) => [...prev, currentDataView]);
+        }
+
+        setDataView(currentDataView);
+      }
+    };
+
+    initDataView();
+  }, [adHocDataViews, dataView, dataViews, ruleParams.name]);
+
+  useEffect(() => {
+    if (!ruleParams.comparator) {
+      preFillProperty('comparator');
+    }
+
+    if (!ruleParams.threshold) {
+      preFillProperty('threshold');
+    }
+
+    if (!ruleParams.timeSize) {
+      preFillProperty('timeSize');
+    }
+
+    if (!ruleParams.timeUnit) {
+      preFillProperty('timeUnit');
+    }
+
+    if (!ruleParams.groupBy) {
+      preFillProperty('groupBy');
+    }
+  }, [
+    preFillProperty,
+    ruleParams.comparator,
+    ruleParams.groupBy,
+    ruleParams.name,
+    ruleParams.threshold,
+    ruleParams.timeSize,
+    ruleParams.timeUnit,
+  ]);
+
+  const onGroupByChange = useCallback(
+    (group: string | null | string[]) => {
+      const gb = group ? (isArray(group) ? group : [group]) : [];
+      setRuleParams('groupBy', gb);
+    },
+    [setRuleParams]
+  );
+
+  const derivedIndexPattern = useMemo<DataViewBase>(
+    () => ({
+      fields: dataView?.fields || [],
+      title: dataView?.getIndexPattern() || 'unknown-index',
+    }),
+    [dataView]
+  );
+
+  const onSelectDataView = useCallback(
+    (newDataView: DataView) => {
+      setDataViewError(undefined);
+      updateProperty('name', newDataView.getIndexPattern());
+      setDataView(newDataView);
+    },
+    [updateProperty]
+  );
+
+  const handleChange = useCallback(
+    (selectedOptions: Array<{ label: string }>) => {
+      const gb = selectedOptions.map((option) => option.label);
+
+      onGroupByChange([...new Set(preselectedOptions.concat(gb))]);
+    },
+    [onGroupByChange, preselectedOptions]
+  );
+
+  const getPreSelectedOptions = () => {
+    return preselectedOptions.map((field) => ({
+      label: field,
+      color: 'lightgray',
+      disabled: true,
+    }));
+  };
+
+  const getUserSelectedOptions = (group: string[] | undefined) => {
+    return (group ?? [])
+      .filter((g) => !preselectedOptions.includes(g))
+      .map((field) => ({
+        label: field,
+      }));
+  };
+
+  const selectedOptions = [...getPreSelectedOptions(), ...getUserSelectedOptions(groupBy)];
+
+  return (
+    <>
+      <EuiTitle size="xs">
+        <h5>
+          <FormattedMessage
+            id="xpack.datasetQuality.rule.alertCondition"
+            defaultMessage="Alert condition"
+          />
+        </h5>
+      </EuiTitle>
+
+      <EuiSpacer size="m" />
+
+      <DataViewSelectPopover
+        dependencies={{ dataViews, dataViewEditor }}
+        dataView={dataView}
```

Comment:
We are not checking here that the dataView contains dataStreams, is this ok @LucaWintergerst, @flash1293?
The rule as it is right now will work for any index.

Reply:
I think this is fine, but maybe Joe has other thoughts

Reply:
No other thoughts as long as the rest of the rule can handle it